sys/proc.c

```diff
@@ -298,8 +301,10 @@ int proc_sendsig(pid_t pid, signo_t sig) {
   }
 
   WITH_MTX_LOCK (&pgrp->pg_lock) {
-    TAILQ_FOREACH (target, &pgrp->pg_members, p_pglist)
+    TAILQ_FOREACH (target, &pgrp->pg_members, p_pglist) {
+      proc_lock(target);
```

Comment:
This looks horrible. I guess `sig_kill` should assume that `proc_lock` is held for the duration of the procedure.

Reply:
yes